samples/webassembly/Triangle/Triangle.cpp

```diff
@@ -0,0 +1,254 @@
+  }
+
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_COMPAT_PROFILE);
+  glfwWindowHint(GLFW_VISIBLE, true);
+  glfwWindowHint(GLFW_DOUBLEBUFFER, true);
+  glfwWindowHint(GLFW_SRGB_CAPABLE, true);
+  glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
+  glfwWindowHint(GLFW_RESIZABLE, GLFW_TRUE);
+
+  GLFWwindow* window = glfwCreateWindow(800, 600, "WebGL Triangle", nullptr, nullptr);
+
+  if (!window) {
+    glfwTerminate();
+    return false;
+  }
+
+  glfwSetErrorCallback([](int error, const char* description) {
+    printf("GLFW Error (%i): %s\n", error, description);
+  });
+
+  glfwSetKeyCallback(window, [](GLFWwindow* window, int key, int, int action, int) {
+    if (key == GLFW_KEY_ESCAPE && action == GLFW_PRESS) {
+      glfwSetWindowShouldClose(window, GLFW_TRUE);
+    }
+  });
+
+  // @lint-ignore CLANGTIDY
+  glfwSetWindowSizeCallback(window, [](GLFWwindow* /*window*/, int width, int height) {
+    printf("Window resized! width=%d, height=%d\n", width, height);
+    width_ = width;
+    height_ = height;
+  });
+
+  glfwGetWindowSize(window, &width_, &height_);
+
+  if (outWindow) {
+    *outWindow = window;
+  }
+
+  return true;
+}
+
+static void initIGL() {
+  // create a device
+  {
+    auto ctx =
+        std::make_unique<igl::opengl::webgl::Context>(igl::opengl::RenderingAPI::GLES3, "#canvas");
+    device_ = std::make_unique<igl::opengl::webgl::Device>(std::move(ctx));
+
+    IGL_ASSERT(device_);
+  }
+
+  // Command queue: backed by different types of GPU HW queues
+  CommandQueueDesc desc{CommandQueueType::Graphics};
+  commandQueue_ = device_->createCommandQueue(desc, nullptr);
+
+  // Color attachment
+  renderPass_.colorAttachments[0] = igl::RenderPassDesc::ColorAttachmentDesc{};
+  renderPass_.colorAttachments[0].loadAction = LoadAction::Clear;
+  renderPass_.colorAttachments[0].storeAction = StoreAction::Store;
+  renderPass_.colorAttachments[0].clearColor = {1.0f, 1.0f, 1.0f, 1.0f};
+  renderPass_.depthAttachment.loadAction = LoadAction::DontCare;
+}
+
+static void createRenderPipeline() {
+  if (renderPipelineState_Triangle_) {
+    return;
+  }
+
+  IGL_ASSERT(framebuffer_);
+
+  RenderPipelineDesc desc;
+
+  desc.targetDesc.colorAttachments.resize(1);
+
+  // @fb-only
+  if (framebuffer_->getColorAttachment(0)) {
+    desc.targetDesc.colorAttachments[0].textureFormat =
+        framebuffer_->getColorAttachment(0)->getFormat();
+  }
+
+  if (framebuffer_->getDepthAttachment()) {
+    desc.targetDesc.depthAttachmentFormat = framebuffer_->getDepthAttachment()->getFormat();
+  }
+
+  desc.shaderStages = ShaderStagesCreator::fromModuleStringInput(
+      *device_, codeVS.c_str(), "main", "", codeFS, "main", "", nullptr);
+  renderPipelineState_Triangle_ = device_->createRenderPipeline(desc, nullptr);
+  IGL_ASSERT(renderPipelineState_Triangle_);
+}
+
+static std::shared_ptr<ITexture> getNativeDrawable() {
+  Result ret;
```

Comment:
Perhaps move these 2 lines below assertion check, and merge drawable declaration and initialization to a single line.